src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -145,18 +144,41 @@ private static async Task<(bool shouldDisplay, bool containsClassExpression)> Sh
                     }
                 }
 
-                // If expression contains this or base keywords then we do not want to refactor call sites
-                // that are not overloads/trampolines because we do not know if the class specific information
-                // is available in other documents.
-                if (variable.RawKind == syntaxKinds.ThisExpression || variable.RawKind == syntaxKinds.BaseExpression)
+                var operation = semanticModel.GetOperation(variable, cancellationToken);
+                if (operation is null || containsClassSpecificStatement)
                 {
-                    containsClassSpecificStatement = true;
+                    continue;
                 }
+
+                // If expression contains this or base keywords, implicitly or explicitly,
+                // then we do not want to refactor call sites that are not overloads/trampolines
+                // because we do not know if the class specific information is available in other documents.
+                containsClassSpecificStatement = OperationOrChildIsInstanceReference(operation, operation.Children);
             }
 
             return (true, containsClassSpecificStatement);
         }
 
+        private static bool OperationOrChildIsInstanceReference(IOperation operation, IEnumerable<IOperation> children)
```

Comment:
use an OperationWalker.  it's faster and avoids an allocation with each child.Children call.

Reply:
changed